### ISSUE TITLE
[NBug] External program returned non-zero exit code.

### ISSUE THREAD

**Issue by eng-marani** [gitextensions/gitextensions#12077]
### Environment

- Git Extensions 5.1.1.17970
- Build 939c0edba68f0c15b8e5180ca06a92c5ce534501
- Git 2.47.0.windows.2
- Microsoft Windows NT 10.0.19045.0
- .NET 8.0.11
- DPI 96dpi (no scaling)
- Portable: False

### Issue description

Exit code: 128
Command: C:\Program Files\Git\bin\git.exe
Arguments: remote
Working directory: P:\GIT\RDC13\CAD\

```
GitExtensions.Extensibility.ExternalOperationException: External program returned non-zero exit code.
 ---> System.Exception: External program returned non-zero exit code.
   --- End of inner exception stack trace ---
   at GitCommands.ExecutableExtensions.ExecuteAsync(IExecutable executable, ArgumentString arguments, Action`1 writeInput, Encoding outputEncoding, CommandCache cache, String extraCacheKey, Boolean stripAnsiEscapeCodes, Boolean throwOnErrorExit, CancellationToken cancellationToken) in C:\projects\gitextensions\src\app\GitCommands\Git\ExecutableExtensions.cs:line 338
   at Microsoft.VisualStudio.Threading.JoinableTask.CompleteOnCurrentThread()
   at Microsoft.VisualStudio.Threading.JoinableTask`1.CompleteOnCurrentThread()
   at Microsoft.VisualStudio.Threading.JoinableTaskFactory.Run[T](Func`1 asyncMethod, JoinableTaskCreationOptions creationOptions)
   at Microsoft.VisualStudio.Threading.JoinableTaskFactory.Run[T](Func`1 asyncMethod)
   at GitCommands.ExecutableExtensions.Execute(IExecutable executable, ArgumentString arguments, Action`1 writeInput, Encoding outputEncoding, CommandCache cache, Boolean stripAnsiEscapeCodes, Boolean throwOnErrorExit, CancellationToken cancellationToken) in C:\projects\gitextensions\src\app\GitCommands\Git\ExecutableExtensions.cs:line 266
   at GitCommands.GitModule.GetRemoteNames() in C:\projects\gitextensions\src\app\GitCommands\Git\GitModule.cs:line 1996
   at GitExtensions.Plugins.GitHub3.GitHub3Plugin.GetHostedRemotes()+MoveNext() in C:\projects\gitextensions\src\plugins\GitHub3\GitHub3Plugin.cs:line 320
   at System.Linq.Enumerable.<Any>g__WithEnumerator|36_0[TSource](IEnumerable`1 source)
   at System.Linq.Enumerable.Any[TSource](IEnumerable`1 source)
   at GitExtensions.Plugins.GitHub3.GitHub3Plugin.GitModuleIsRelevantToMe() in C:\projects\gitextensions\src\plugins\GitHub3\GitHub3Plugin.cs:line 305
   at GitUI.PluginRegistry.<>c.<TryGetGitHosterForModule>b__16_0(IRepositoryHostPlugin gitHoster) in C:\projects\gitextensions\src\app\GitUI\Plugin\PluginRegistry.cs:line 104
   at System.Linq.Enumerable.TryGetFirst[TSource](IEnumerable`1 source, Func`2 predicate, Boolean& found)
   at System.Linq.Enumerable.FirstOrDefault[TSource](IEnumerable`1 source, Func`2 predicate)
   at GitUI.PluginRegistry.TryGetGitHosterForModule(IGitModule module) in C:\projects\gitextensions\src\app\GitUI\Plugin\PluginRegistry.cs:line 104
   at GitUI.CommandsDialogs.RevisionDiffControl.RegisterGitHostingPluginInBlameControl() in C:\projects\gitextensions\src\app\GitUI\CommandsDialogs\RevisionDiffControl.cs:line 1502
   at GitUI.CommandsDialogs.FormBrowse.SetGitModule(Object sender, GitModuleEventArgs e) in C:\projects\gitextensions\src\app\GitUI\CommandsDialogs\FormBrowse.cs:line 1734
   at GitUI.CommandsDialogs.FormBrowse.SetWorkingDir(String path, ObjectId selectedId, ObjectId firstId) in C:\projects\gitextensions\src\app\GitUI\CommandsDialogs\FormBrowse.cs:line 1670
   at GitUI.LeftPanel.SubmoduleNode.Open() in C:\projects\gitextensions\src\app\GitUI\LeftPanel\SubmoduleNode.cs:line 71
   at GitUI.LeftPanel.SubmoduleNode.OnDoubleClick() in C:\projects\gitextensions\src\app\GitUI\LeftPanel\SubmoduleNode.cs:line 120
   at GitUI.LeftPanel.RepoObjectsTree.<>c.<OnNodeDoubleClick>b__68_0(Node node) in C:\projects\gitextensions\src\app\GitUI\LeftPanel\RepoObjectsTree.cs:line 667
   at GitUI.LeftPanel.Node.OnNode[T](TreeNode treeNode, Action`1 action) in C:\projects\gitextensions\src\app\GitUI\LeftPanel\Node.cs:line 114
   at GitUI.LeftPanel.RepoObjectsTree.OnNodeDoubleClick(Object sender, TreeNodeMouseClickEventArgs e) in C:\projects\gitextensions\src\app\GitUI\LeftPanel\RepoObjectsTree.cs:line 667
   at System.Windows.Forms.TreeView.WndProc(Message& m)
   at System.Windows.Forms.NativeWindow.Callback(HWND hWnd, MessageId msg, WPARAM wparam, LPARAM lparam)
```

After creating a different working tree for another branch, all submodules become corrupted, and starting to switch to submodule produces an eerror

### Steps to reproduce

After creating a new working tree from a checkout having some submodules, the new working tree has all submodule corrupted, and you cant manage them without crashes

**Comment by mstv** [comment]
An `ExternalOperationException` - particarly "External program returned non-zero exit code" - is not a crash. It just gives you the info that git itself was unhappy. 
It seems that the GitHub3Plugin lets bubble up this exception.
Consider deactivating it as long as your config is broken.

**Comment by eng-marani** [comment]
I have solved adding this to my .gitmodules:

[submodule "CAD"]
	path = CAD
	url = https://bitbucket.org/software-rd/xxxxxxx.git
	branch = master
	**ignore = all**
[submodule "COMMONS"]
	path = COMMONS
	url = https://bitbucket.org/software-rd/yyyyyyy.git
	branch = master
	ignore = all

Dont ask me why but without this Ignore=all submodules got stuck cloning back to proper folder and takes forever to complete. With this it was immediate.... i've found it by pure lucky googling around. Do this gives you any clue ?
Basically, there are something screwed up when creating new workingtree from gitextensions, it will always stuck cloning submodules unless i add this ignore=all

**Comment by mstv** [comment]
These additional entries have not been created by the git commands which GE (usually) executes.

I have just created a new worktree of the GE repo. Its `.gitmodules` just contains:
```
[submodule "Externals/Git.hub"]
        path = Externals/Git.hub
        url = ../../gitextensions/Git.hub.git
[submodule "Externals/conemu-inside"]
        path = Externals/conemu-inside
        url = ../../gitextensions/conemu-inside.git
[submodule "Externals/ICSharpCode.TextEditor"]
        path = Externals/ICSharpCode.TextEditor
        url = ../../gitextensions/ICSharpCode.TextEditor.git
[submodule "Externals/EasyHook"]
        path = Externals/EasyHook
        url = ../../gitextensions/EasyHook.git
```
which makes the worktree use shared `.git` folders for the submodules, too.

After creating the worktree, the submodules need to be initialized.

![Image](https://github.com/user-attachments/assets/a18aeaf6-bddb-47d9-b839-91873b3bafca)
